### PR TITLE
[86a3mzx2k] Keycloak helm spec

### PR DESCRIPTION
**Pull request by deepankermishra**

### CODE REVIEW

File: charts/dynamoai-api/templates/deployment.yaml
Comment (brarkaran) [inline charts/dynamoai-api/templates/deployment.yaml:1]: We need changes in the helpers.tpl file to define these environment variables separately, and add them to the correct place in dynamoai-core

File: charts/dynamoai-api/templates/deployment.yaml [hunk 1]
```diff
@@ -48,6 +48,14 @@ spec:
...
             {{- end }}
+            {{- if eq .Values.api.authProvider "keycloak" }}
+              {{- with .Values.api.keycloakEnv }}
+                {{- toYaml . | nindent 12 }}
+              {{- end }}
+            {{- else if eq .Values.api.authProvider "cognito" }}
+              {{- with .Values.api.cognitoEnv }}
+                {{- toYaml . | nindent 12 }}
+              {{- end }}
           {{- if .Values.global.config.proxy }}
```
Comment (ishucr7) [inline charts/dynamoai-api/templates/deployment.yaml:59]: add: minio envs

File: charts/dynamoai-api/values.yaml [hunk 2]
```diff
@@ -94,6 +94,32 @@ api:
...
+    - name: 'API_AUTH_MODE'
+      value: 'keycloak-and-key'
+
+  cognitoEnv:
+    - name: 'API_AUTH_MODE'
+      value: 'cognito-and-key'
+
+  storageProvider: minio # or s3
+  minioEnv:
+    - name: S3_ENDPOINT
```
Comment (ishucr7) [inline charts/dynamoai-api/values.yaml:117]: Can we add another variable here `AWS_ENDPOINT_URL_S3`?
It's value will be same as S3_ENDPOINT and we might deprecate S3_ENDPOINT (I'm trying that out)

File: charts/dynamoai-iam/values.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      nginx.ingress.kubernetes.io/rewrite-target: /$1
+    hostname: dev.dynamofl.com
+
+  keycloakConfigCli:
+    enabled: true
+    existingConfigmap: "keycloak-config"
+
+  extraEnvVars:
+    - name: KEYCLOAK_EXTRA_ARGS
+      value: "--hostname-url=https://dev.dynamofl.com/keycloak --hostname-admin-url=https://dev.dynamofl.com/keycloak"
```
Comment (ishucr7) [inline charts/dynamoai-iam/values.yaml:33]: We need to add --import-realm here

**Review by ishucr7 — COMMENTED** [review]
LGTM. didn't find minio in this, is that in a separate PR?

**Pull request merged**